src/main/java/com/github/harti2006/neo4j/StartNeo4jServerMojo.java

```diff
@@ -120,8 +136,52 @@ private void startNeo4jServer() throws MojoExecutionException {
             } else {
                 throw new MojoExecutionException("Neo4j server did not start up properly");
             }
+            
+            // Now we need to wait it replies
+            checkServerReady ();
+            
+            // Went up! If it's a new DB, let's change the password.
+            setNewPassword ();
         } catch (IOException | InterruptedException e) {
-            throw new MojoExecutionException("Could not start neo4j server", e);
+            throw new MojoExecutionException( "Could not start neo4j server", e);
         }
     }
+    
+    private void checkServerReady () throws MojoExecutionException, InterruptedException
+    {
+    		String pwd = deleteDb ? "neo4j" : password;
```

Comment:
At this point, the password is always the initial one "neo4j", because `setNewPassword` will only be executed after the successful help check.

Reply:
It doesn't seem that to me. deleteDb is a paramter that tells the plugin to wipe out any existing DB and create a new one (which will have the default pwd). If that's false, the old DB will be started untouched (in my opinion, this can be useful to restart the server manually, after testing, for manua inspection). So, in the latter case this is supposed to have the password passed by the password parameter.

Reply:
You are right, I missed the point where the plugin/server can be restarted multiple times without wiping the data.